modules/mapbox/src/deck-utils.ts

```diff
@@ -1,22 +1,40 @@
 import {Deck, WebMercatorViewport, MapView} from '@deck.gl/core';
-
-export function getDeckInstance({map, gl, deck}) {
+import type {DeckProps, MapViewState, Layer} from '@deck.gl/core';
+import type MapboxLayer from './mapbox-layer';
+import type {Map} from 'mapbox-gl';
+
+type UserData = {
+  isExternal: boolean;
+  currentViewport?: WebMercatorViewport | null;
+  mapboxLayers: Set<MapboxLayer<any>>;
+  mapboxVersion: {minor: number; major: number};
+};
+
+export function getDeckInstance({
+  map,
+  gl,
+  deck
+}: {
+  map: Map;
```

Comment:
```suggestion
  map: Map & {__deck?: Deck | null};
```

The annotation is repeated in many places, I would say it is better to extend the `Map` type.